phosphobot/models.py

```diff
@@ -1060,6 +1076,157 @@ def save(self, meta_folder_path: str) -> None:
+
+        # Compute the sum and square sum for each column
+        column_sums = {
+            col: {
+                "sum": np.sum(np.array(deleted_episode_df[col].tolist()), axis=0),
+                "square_sum": np.sum(
+                    np.array(deleted_episode_df[col].tolist()) ** 2, axis=0
+                ),
+            }
+            for col in deleted_episode_df.columns
+        }
+        logger.info(f"Column sums: {column_sums}")
+        # Update stats for each field in the StatsModel
+        for field_name, field in StatsModel.model_fields.items():
+            # TODO task_index is not updated since we do not support multiple tasks
+            # observation_images has a special treatment
+            if field_name in ["task_index", "observation_images"]:
+                continue
+            logger.info(f"Updating field {field_name}")
+            # Get the field value from the instance
+            field_value = getattr(self, field_name)
+            # Convert observation_state to observation.state
+            field_name = (
+                "observation.state" if field_name == "observation_state" else field_name
+            )
+            # Update statistics
+            if field_name in column_sums:
+                # Maximum of debug info
+                logger.info(f"Column sums for {field_name}: {column_sums[field_name]}")
+                logger.info(f"Field value: {field_value}")
+                logger.info(f"Field value mean: {field_value.mean}")
+
+                # Subtract sums
+                field_value.sum -= column_sums[field_name]["sum"]
+                field_value.square_sum -= column_sums[field_name]["square_sum"]
+
+                logger.info(f"Field value sum: {field_value.sum}")
+
+                # Update count
+                field_value.count -= nb_steps_deleted_episode
+
+                # Recalculate mean and standard deviation
+                if field_value.count > 0:
+                    field_value.mean = field_value.sum / field_value.count
+                    field_value.std = np.sqrt(
+                        (field_value.square_sum / field_value.count)
+                        - np.square(field_value.mean)
+                    )
+
+        # For image we need to load the mp4 video
+        logger.info("Updating stats for images")
+        # Extract the episode index from the parquet file name
+        episode_index = int(parquet_path.split("_")[-1].split(".")[0])
+
+        # List cameras_folder:
+        folder_videos_path = (
+            "/".join(parquet_path.split("/")[:-3]) + "/videos/chunk-000"
+        )
+
+        cameras_folder = os.listdir(folder_videos_path)
+        for camera_folder in cameras_folder:
+            # Create the path of the video episode_{episode_index:06d}.mp4
+            video_path = os.path.join(
+                folder_videos_path, camera_folder, f"episode_{episode_index:06d}.mp4"
+            )
```

Comment:
way too error prone way to get the cameras folder... please change the function parameters instead